lib/icontrol_rest/client.rb

```diff
@@ -64,38 +64,51 @@ def logger
     #         "nameServers"=>["1.2.3.72", "1.2.3.73"],
     #         "search"=>["domain.com"]}
     #
-    %w[delete get].each do |action|
-      define_method(action.to_s) do |route, _arg = nil|
-        send_request { self.class.send(action, route, @options) }
+    %w[delete get post put patch].each do |action|
```

Comment:
Does this not duplicate what is below?

Reply:
Oh, darn. I tried to combine these but it didn't work out. I'll fix it.